indent/ruby.vim

```diff
@@ -580,10 +608,19 @@ function! s:IndentingKeywordInMSL(msl_info)
   " If the MSL line had an indenting keyword in it, add a level of indent.
   " TODO: this does not take into account contrived things such as
   " module Foo; class Bar; end
-  if s:Match(info.plnum_msl, s:ruby_indent_keywords)
+  let col = s:Match(info.plnum_msl, s:ruby_indent_keywords)
+  if col > 0
     let ind = indent(info.plnum_msl) + info.sw
     if s:Match(info.plnum_msl, s:end_end_regex)
       let ind = ind - info.sw
+    elseif s:IsAssignment(getline(info.plnum_msl), col)
+      if g:ruby_indent_end_alignment == 'hanging'
+        " hanging indent
+        let ind = col + info.sw - 1
+      else
+        " align with variable
+        let ind = indent(info.plnum_msl) + info.sw
+      endif
```

Comment:
Hmmm. This one doesn't cause any of the existing tests to fail.

I've come up with two test cases for it though. I'll add them to this PR.

"Hanging" indent (the alignment is wonky for the chained methods):
```ruby
      assert_correct_indenting <<-EOF
        variable = if [].
            map { |x| x * 2 }.
            filter { |x| x % 3 == 0 }.
            empty?
                     something
                   end
      EOF
```

"Variable" indent:
```ruby
      assert_correct_indenting <<-EOF
        variable = if [].
            map { |x| x * 2 }.
            filter { |x| x % 3 == 0 }.
            empty?
          something
        end
      EOF
```